src/platform/i386/chal_pgtbl.c

```diff
@@ -263,12 +263,41 @@ chal_pgtbl_deactivate(struct captbl *t, struct cap_captbl *dest_ct_cap, unsigned
 	return ret;
 }
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wreturn-local-addr"
```

Comment:
Wow, I've never seen this. Interesting! Can you add a comment for why it is necessary here so that we know in the future when/if we can remove it?